test/functional/ex_cmds/print_commands_spec.lua

```diff
@@ -0,0 +1,11 @@
+local helpers = require('test.functional.helpers')(after_each)
+local eq, command, funcs = helpers.eq, helpers.command, helpers.funcs
+
+describe(':z^', function()
+  before_each(clear)
+
+  it('correctly sets the cursor after :z^', function()
+    command('silent! z^')
```

Comment:
Before the change `z^` was incorrect. It was absolutely valid to keep anything which made test error out, fail (busted differentiates between error and failure), crash, hang forever, do whatever else except yielding success status.